### PR TITLE
[WIP] Issue #494: Add browser-icon module and icons code

### PR DESCRIPTION
**Pull request by jonalmeida**
This is the initial conversion of browser-icons [from fennec][0] to a component. It's still in Java and will be converted to Kotlin incrementally to make reviewing easier.

For now, we would need to commit a giant code change..

[0]: https://dxr.mozilla.org/mozilla-central/source/mobile/android/base/java/org/mozilla/gecko/icons

### CODE REVIEW

File: components/browser/icons/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# [Android Components](../../../README.md) > Browser > Icons
+
+Favicons support for Android browser apps.
```
Comment (pocmo) [inline components/browser/icons/README.md]: nit: Not only `favicons` :)

And: What does it actually do? :)

File: components/browser/icons/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Usage
+
+### Setting up the dependency
+
+Use gradle to download the library from JCenter:
```
Comment (pocmo) [inline components/browser/icons/README.md]: JCenter :)

File: components/browser/icons/build.gradle [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    testImplementation Deps.testing_mockito
+}
+
+archivesBaseName = "icons"
+
+apply from: '../../../publish.gradle'
+ext.configurePublish(
+        'org.mozilla.components',
+        'icons',
+        'Favicons support for Android browser apps.')
```
Comment (pocmo) [inline components/browser/icons/build.gradle]: Old config!

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/IconCallback.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
```
Comment (pocmo) [inline components/browser/icons/src/main/java/mozilla/components/browser/icons/IconCallback.java]: Those files need more license headers 🙃

**Comment by jonalmeida** [comment]
Closing this for now to avoid cluttering our PR list.

**Pull request closed without merging**